source/velocity_boundary_conditions/interface.cc

```diff
@@ -52,6 +52,31 @@ namespace aspect
 
 
     template <int dim>
+    Tensor<1,dim>
+    Interface<dim>::boundary_velocity (const Point<dim> &) const
+    {
+      // we can only get here if the new-style function calls it; this means that
+      // the derived class did not override the new-style function, and because
+      // we are here, it also did not override the current old-style function
+      Assert (false, ExcMessage ("A derived class needs to override either the "
+                                 "boundary_velocity() or boundary_velocity(types::boundary_id) function."));
```

Comment:
this doesn't quite match the actual argument lists of the two functions. Maybe say "one of the two boundary_velocity() functions"?